File: src/main/java/de/blankedv/sx4draw/Position.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package de.blankedv.sx4draw;

import static de.blankedv.sx4draw.Constants.INVALID_INT;

/**
 * @author Michael Blank <mblank@bawue.de>
 */
public class Position {
    public int x; // starting point
    public int y;
    public int x2 = INVALID_INT; // endpoint - x2 always >x
    public int y2 = INVALID_INT;
    public int xt = INVALID_INT; // "thrown" position for turnout
    public int yt = INVALID_INT;
}

File: src/main/java/de/blankedv/sx4draw/Position.kt
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package de.blankedv.sx4draw

import de.blankedv.sx4draw.Constants.INVALID_INT

/**
 * @author Michael Blank <mblank></mblank>@bawue.de>
 */
data class Position (
    var x: Int = 0, // starting point
    var y: Int = 0,
    var x2: Int = INVALID_INT, // endpoint - x2 always >x, closed pos. for turnout
    var y2: Int = INVALID_INT,
    var xt: Int = INVALID_INT, // "thrown" position for turnout
    var yt: Int = INVALID_INT
)
